lib/src/main/java/uk/gov/di/ipv/core/library/service/AccessTokenService.java

```diff
@@ -64,4 +73,23 @@ public void persistAccessToken(AccessTokenResponse tokenResponse, String ipvSess
         accessTokenItem.setIpvSessionId(ipvSessionId);
         dataStore.create(accessTokenItem);
     }
+
+    public ValidationResult<ErrorObject> extractJwt(String requestBody) {
+
+        Map<String, String> stringMap = RequestHelper.parseRequestBody(requestBody);
+
+        try {
+            SignedJWT str = SignedJWT.parse(String.valueOf(stringMap.get("client_assertion")));
+
+            JWTClaimsSet claimsSet = str.getJWTClaimsSet();
+
+            if (claimsSet != null) {
+                return ValidationResult.createValidResult();
+            }
+        } catch (ParseException e) {
+            LOGGER.error("Unable to parse Claims set {} ", e.getMessage());
+            return new ValidationResult<>(false, OAuth2Error.INVALID_REQUEST_OBJECT);
```

Comment:
There is an OAuth2Error.INVALID_CLIENT error object which might be better suited here?